gapis/api/gles/api/draw_commands.api

```diff
@@ -183,23 +190,46 @@ sub void DrawElementsBaseVertex(GLenum draw_mode, GLsizei indices_count, GLenum
   DrawElements(ctx, draw_mode, indices_count, indices_type, indices, 1, base_vertex)
 }
 
+// See GLES 3.2 spec '10.5 Drawing Commands Using Vertex Arrays'
+class DrawElementsIndirectCommand {
+  u32 count
+  u32 instanceCount
+  u32 firstIndex
+  s32 baseVertex
+  u32 reservedMustBeZero
+}
+
 @if(Version.GLES31)
 @draw_call(isDrawCall())
 @transform_feedback(isTransformFeedback())
 @doc("https://www.khronos.org/opengles/sdk/docs/man31/html/glDrawElementsIndirect.xhtml", Version.GLES31)
 @doc("https://www.khronos.org/opengles/sdk/docs/man32/html/glDrawElementsIndirect.xhtml", Version.GLES32)
 cmd void glDrawElementsIndirect(GLenum draw_mode, GLenum indices_type, const void* indirect) {
-  checkPrimitiveType(draw_mode)
   checkIndicesType(indices_type)
-
-  ctx := GetContext()
   CheckNonDefaultVertexArrayBound()
   // TODO: Error if enabled array is not bound or is mapped.
+  ctx := GetContext()
   if ctx.Bound.DrawIndirectBuffer == null { glErrorInvalidOperation() }
   // TODO: INVALID_OPERATION error if the command would source data beyond the end of the buffer object.
   // TODO: CheckEQ(as!u64(indirect) % 4, 0)
+
+  // TODO: The following is the actual sepc implementation, yet it does not compile because of an seemly impossible cast.
+  //
+  // command := as!DrawElementsIndirectCommand*(indirect)[0]
+  // size_of_type := switch(indices_type) {
+  //   case GL_UNSIGNED_BYTE:  { 1 }
+  //   case GL_UNSIGNED_SHORT: { 2 }
+  //   case GL_UNSIGNED_INT:   { 4 }
+  // }
+  //
+  // ## The cast as!void*() below fails:
+  // DrawElementsInstancedBaseVertex(draw_mode, as!GLsizei(command.count), indices_type, as!void*(command.firstIndex * as!u32(size_of_type)), as!GLsizei(command.instanceCount), as!GLint(command.baseVertex))
+
+  // TODO: once the cast issue above is fixed, we should be able to get rid of these three calls:
+  checkPrimitiveType(draw_mode)
   WriteGPUFramebufferData(ctx)
-  DrawCommandDependencies(ctx)
+  DrawCommandDependencies(ctx, as!DrawElementsIndirectCommand*(indirect)[0].count, as!DrawElementsIndirectCommand*(indirect)[0].instanceCount)
```

Comment:
Here we cannot abstract `as!DrawElementsIndirectCommand*(indirect)[0]` in a local variable because it triggers a compilation error down the line. The error is due to the fact that DrawCommandDependencies is marked as `@spy_disabled`, so in spy-related generated code, the local variable appears to be never used, which leads to a compilation error.

Reply:
Does this work?
```
foo := as!DrawElementsIndirectCommand*(indirect)[0].count
_ = foo
``

Reply:
I works, thanks!